internal/host/host.go

```diff
@@ -319,27 +319,32 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
 	if err != nil {
 		return err
 	}
-	if h.ClusterID != nil && h.ClusterID.String() != "" {
-		if cluster == nil {
-			cluster, err = common.GetClusterFromDB(m.db, *h.ClusterID, common.UseEagerLoading)
-			if err != nil {
-				log.WithError(err).Errorf("not updating inventory - failed to find cluster %s", h.ClusterID.String())
-				return common.NewApiError(http.StatusNotFound, err)
-			}
-		}
 
-		err = m.populateDisksEligibility(ctx, inventory, cluster, h)
+	if h.ClusterID != nil && h.ClusterID.String() != "" {
+		cluster, err = common.GetClusterFromDB(m.db, *h.ClusterID, common.UseEagerLoading)
 		if err != nil {
-			log.WithError(err).Errorf("not updating inventory - failed to check disks eligibility for host %s", h.ID)
-			return common.NewApiError(http.StatusInternalServerError, err)
+			log.WithError(err).Errorf("not updating inventory - failed to find cluster %s", h.ClusterID.String())
+			return common.NewApiError(http.StatusNotFound, err)
 		}
-		m.populateDisksId(inventory)
+	}
 
-		inventoryStr, err = common.MarshalInventory(inventory)
-		if err != nil {
-			return err
-		}
+	infraEnv, err := common.GetInfraEnvFromDB(m.db, h.InfraEnvID)
+	if err != nil {
+		log.WithError(err).Errorf("not updating inventory - failed to find infra env %s", h.InfraEnvID.String())
+		return common.NewApiError(http.StatusNotFound, err)
+	}
```

Comment:
Do we need to run this code if the host is bound (if the cluster variable got set above)?

Reply:
the flow is that we get all the data here ( cluster, infra env), and let the DiskIsEligble to decide how to check eligibility